### DISCUSSION TITLE
Feature Suggestions/Ideas

### DISCUSSION THREAD

**Discussion by tboy1337** [masterking32/MasterCryptoFarmBot#9] [Ideas]
1) Asynchronous/parallel running of accounts - one by one approach is ok with a small number of accounts but the more accounts you have the worse it is.

2) Use an .ini filetype for the config (e.g. `import configparser`) and build an executable for the bot - an executable would be really useful as python and the project dependencies wouldn’t need to be installed (I have a github actions workflow that might be useful to you [Build & Release.yml](https://gist.github.com/tboy1337/caa92794ba1120d3501b1a9625081f60#file-build-release-yml), it isnt fully finished as it needs to point to the .spec file not the project .py files, here’s an [example.spec](https://gist.github.com/tboy1337/81c5ac37727df862dbed531484fd1a89#file-example-spec), Ive also already created a [config.ini.example](https://gist.github.com/tboy1337/63054d1de4c73190876b937c62de45d4#file-config-ini-example)).

3) Don’t make GIT the main way to update project - its another dependency that ideally isn’t needed (the best part is no part), if you build an executable you could use [tufup](https://github.com/dennisvang/tufup).

4) More anti-bot detection functionality - for example an optional period of "sleeping" (4-5 hours), another example is when all playground games have been played for the day the timer needs to be made larger as there is now less time between each run due to the smaller amount of activity per each account (I hope that makes sense lol).

5) Local log files - local log files would be very useful to someone who doesn’t use telegram regularly, someone who doesn’t use their phone much or someone who doesn’t have a smartphone.

6) Use API_ID and API_HASH authentication method instead of Bearer token - as I understand it the Bearer token can change but the API_ID and API_HASH will stay the same, this seems like a better system.

**Comment by marrsripon** [comment]
May be using both system will be useful login via API or Bearer Token . There may be action of choose any one from beginning
